src/webRTCEndpoint.ts

```diff
@@ -373,22 +380,31 @@ export class WebRTCEndpoint extends (EventEmitter as new () => TypedEmitter<
       case "connected":
         this.localEndpoint.id = deserializedMediaEvent.data.id;
 
+
         this.emit(
           "connected",
           deserializedMediaEvent.data.id,
           deserializedMediaEvent.data.otherEndpoints
         );
 
+
+
         let otherEndpoints = deserializedMediaEvent.data
           .otherEndpoints as Endpoint[];
         otherEndpoints.forEach((endpoint) => {
           this.addEndpoint(endpoint);
         });
 
+
         otherEndpoints.forEach((endpoint) => {
+          const trackIdToSimulcastConfig = new Map<String, SimulcastConfig>(
+            Object.entries(endpoint.trackIdToSimulcastConfig)
+          )
+
           Array.from(endpoint.trackIdToMetadata.entries()).forEach(
             ([trackId, metadata]) => {
-              const ctx = new TrackContextImpl(endpoint, trackId, metadata);
+              const simulcastConfig: SimulcastConfig = trackIdToSimulcastConfig.get(trackId) as SimulcastConfig;
+              const ctx = new TrackContextImpl(endpoint, trackId, metadata, simulcastConfig);
```

Comment:
I suggest not using the `as` operator to ignore a possible undefined. I would add an explicit exception.
```suggestion
              const simulcastConfig: SimulcastConfig | undefined = endpoint.trackIdToSimulcastConfig.get(trackId);
              if(simulcastConfig) throw Error(`Simulcast config for track ${trackId} not found`)

              const ctx = new TrackContextImpl(endpoint, trackId, metadata, simulcastConfig);
```